src/assignments/tech-jobs-console.rst

```diff
@@ -0,0 +1,511 @@
+do, and you can easily open, read, and edit them in plain text editors.
+
+Open up ``jobs_data.csv``, which is in the ``resources`` folder at the
+top level of the project. You’ll see that the first line is:
+
+.. sourcecode:: bash
+
+   name,employer,location,position type,core competency
+
+While it isn’t required, the first line of a CSV file often represents
+the column names. We have 5 names here, which indicates that each of our
+rows in the CSV file should have 5 fields. In this file format, a “row”
+corresponds to a new line. So each line below the first will constitute
+a row of data, or a record.
+
+Have a look a the data below line 1, and ask yourself the following
+questions: - Which fields match up with which column names above? - Why
+do some lines/rows (e.g. line 10) have more commas than others, if
+commas are supposed to separate columns? - What role do the
+double-quotes play?
+
+The TechJobs Class
+-------------------
+
+The ``TechJobs`` class contains the ``main`` method that will drive our
+program’s functionality. It contains three methods:
+
+#. ``main`` - The main application runner.
+#. ``getUserSelection`` - A utility method that displays a menu of choices and
+   returns the user’s selection.
+#. ``printJobs`` - This is meant to print a list of jobs to the console in a
+   nicely formatted manner, but hasn’t been implemented yet. This will be part
+   of your job.
+
+Let’s look at each of these.
+
+The ``main`` Method
+^^^^^^^^^^^^^^^^^^^^
+
+The logic within ``main`` presents menus in turn, and based on the
+user’s choice, takes appropriate action.
+
+It begins by declaring two local variables: ``columnChoices`` and
+``actionChoices``. These contain information relating to the menus that
+we’ll display, and we’ll look at them in more detail later.
+
+Next, we notice a ``while loop`` that starts ``while (true)``. This may
+seem odd, but actually, it makes a lot of sense after a short
+explanation. We want our application to continually run until the user
+has decided they want to quit. The simplest way to do this is to loop
+forever. When the user wants to quit, they can kill our program by
+pressing ctrl-C (a widely-known command to kill a console application).
+As you saw above, however, IntelliJ’s Run pane works slightly
+differently and you’ll need to rely on the red “stop” icon to stop the
+program.
+
+The ``main`` method can be summarized as follows:
+
+#. Present the user with choices on how to view data: list or search.
+#. Based on that choice, prompt them with the column that they would
+   like to apply the choice to. In the case of a search, we also ask for
+   a search term.
+#. Carry out the “query” to the ``JobData`` class via one of its public
+   methods.
+#. Display the results of the “query.”
+#. Repeat.
+
+The word “query” is in quotes here because we’re not really carrying out
+a database query, but the net effect is the same as if we were. We ask a
```

Comment:
Have they learned about database queries here? We haven't talked to them about SQL at all.

Reply:
The students have not learned about DBs yet, so I updated the paragraph and preceding list.  Let me know if you think we should keep the new ol below "Repeat", or if we should just remove it altogether.

Reply:
I think it is pretty good! As long as students can do what they need to do without writing SQL, I am good.

Reply:
OK.  Thanks for the check.